tests/ballerina-unit-test/src/test/resources/test-src/workers/flush-workers.bal

```diff
@@ -62,14 +62,19 @@ function flushAll() returns string {
             }
         }
 
-        worker w2 {
+        worker w2 returns error?{
```

Comment:
Why do we always have to return an error? Is it because there is a wait. Even if that is the case, real users will also have to do a hack like you have done here within the worker. IMO we should not have such a requirement.

Reply:
it's not mandatory, but in that case sync send has to be written as `() result = a ->> w2;`. this is due to type system limitation. Will be fixed to `a ->> w2;` when we move interaction analysis before type checker. Until then I made them return errors because that will be the most complex case for sync send and i wanted test that.

Please go ahead and change them back to return nil in the places you need.